<!DOCTYPE html>
<html lang="ja-JP">
<head>

<meta name="robots" content="noindex">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="../style.css">
<link rel="stylesheet" href="../syntax-highlight.css">
<link rel="icon" type="image/png" href="../rurema.png">

<title>module Net::HTTPExceptions (Ruby 3.1.2 リファレンスマニュアル)</title>
<meta name="description" content="HTTP 例外クラスです。">
<script src="../script.js"></script>
</head>
<body>


<header>
  <nav>
    <ol class="inline-breadcrumb-list">
      <li>
        <a href="../doc/index.html">Ruby 3.1.2 リファレンスマニュアル</a>
      </li>
      <li>
        <a href="../library/index.html">ライブラリ一覧</a>
      </li>
      <li>
        <a href="../library/net=2fhttp.html">net/httpライブラリ</a>
      </li>
      <li>
        Net::HTTPExceptionsモジュール
      </li>
    </ol>
  </nav>

<h1>module Net::HTTPExceptions</h1>
</header>

<main>



  <nav>
    クラス・モジュールの継承リスト:
    <ol class="inline-ancestors-list">
      
      <li>
        Net::HTTPExceptions
      </li>
    </ol>
  </nav>











<h2>要約</h2>
<p>
HTTP 例外クラスです。
</p>
<p>
実際にはこれを include した以下のサブクラスの例外が発生します。
</p>
<ul>
<li><a href="../class/-net=3a=3a-h-t-t-p-error.html">Net::HTTPError</a></li>
<li><a href="../class/-net=3a=3a-h-t-t-p-retriable-error.html">Net::HTTPRetriableError</a></li>
<li><a href="../class/-net=3a=3a-h-t-t-p-server-exception.html">Net::HTTPServerException</a></li>
<li><a href="../class/-net=3a=3a-h-t-t-p-fatal-error.html">Net::HTTPFatalError</a></li>
</ul>
<p>
また、例外を発生させるためには <a href="../method/-net=3a=3a-h-t-t-p-response/i/value.html">Net::HTTPResponse#value</a> を呼ぶ必要があります。
</p>



<h2>目次</h2>
<dl>

<dt>インスタンスメソッド</dt>
<dd>
<ul class="class-toc">

<li><a href="#I_RESPONSE">response</a></li>

</ul>
</dd>

</dl>




</dl>


<h2>インスタンスメソッド</h2>
<dl>

<dt class="method-heading" id="I_RESPONSE"><code>response -&gt; Net::HTTPResponse</code><span class="permalink">[<a href="../method/-net=3a=3a-h-t-t-p-exceptions/i/response.html">permalink</a>][<a href="https://docs.ruby-lang.org/en/3.1.2/Net/HTTPExceptions.html#method-i-response">rdoc</a>]</span></dt>
<dd class="method-description">
<p>
例外の原因となったレスポンスオブジェクトを返します。
</p>
<pre class="highlight ruby">
<span class="caption">例</span>
<code>
<span class="nb">require</span> <span class="s1">'net/http'</span>

uri <span class="o">=</span> <span class="s2">"</span><span class="s2">http://www.example.com/invalid.html</span><span class="s2">"</span>
response <span class="o">=</span> <span class="no">Net</span><span class="o">::</span><span class="no">HTTP</span><span class="p">.</span><span class="nf">get_response</span><span class="p">(</span><span class="no">URI</span><span class="p">.</span><span class="nf">parse</span><span class="p">(</span>uri<span class="p">)</span><span class="p">)</span>
<span class="k">begin</span>
  response<span class="p">.</span><span class="nf">value</span>
<span class="k">rescue</span> <span class="o">=&gt;</span> e
  e<span class="p">.</span><span class="nf">response</span> <span class="c1"># =&gt; #&lt;Net::HTTPNotFound 404 Not Found readbody=true&gt;
</span><span class="k">end</span>
</code></pre>
</dd>


</dl>

</main>

<footer id="footer">
  <a rel="license" href="https://creativecommons.org/licenses/by/3.0/">
    <img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/3.0/88x31.png" />
  </a>

  <a href="https://github.com/rurema/doctree/issues/new" id="feedback-link">フィードバックを送る</a>
  
  <script>if (window.URLSearchParams) { document.getElementById("feedback-link").search = new URLSearchParams({'body': document.location}); }</script>
</footer>
</body>
</html>
